Kickstarter-iOS/DataSources/PledgeDataSource.swift

```diff
@@ -10,26 +10,80 @@ final class PledgeDataSource: ValueCellDataSource {
     case summary
   }
 
-  func load(amount: Double, currency: String, delivery: String, isLoggedIn: Bool) {
+  enum PledgeInputRow {
+    case pledgeAmount(amount: Double, currency: String)
+    case shippingLocation(location: String, amount: Double, currencyCode: String)
+
+    var isShipping: Bool {
+      switch self {
+      case .shippingLocation: return true
+      default: return false
+      }
+    }
+
+    var isPledgeAmount: Bool {
+      switch self {
+      case .pledgeAmount: return true
+      default: return false
+      }
+    }
+  }
+
+  func load(data: PledgeTableViewData) {
+    self.loadProjectSection(delivery: data.delivery)
+
+    self.loadInputsSection(amount: data.amount, currency: data.currency, currencyCode: data.currencyCode,
+                           requiresShippingRules: data.requiresShippingRules)
+
+    self.loadSummarySection(isLoggedIn: data.isLoggedIn)
+  }
+
+  func loadSelectedShippingRule(data: SelectedShippingRuleData) {
+    guard self.numberOfItems(in: PledgeDataSource.Section.inputs.rawValue) > 1 else { return }
+
+    self.set(value: PledgeInputRow.shippingLocation(location: data.location, amount: data.amount,
+                                                    currencyCode: data.currencyCode),
+             cellClass: PledgeShippingLocationCell.self,
+             inSection: Section.inputs.rawValue,
+             row: 1)
+  }
+
+  func shippingCellIndexPath() -> IndexPath? {
+    let inputsRowCount = self.numberOfItems(in: PledgeDataSource.Section.inputs.rawValue)
+
+    guard inputsRowCount > 0 else { return nil }
+
+    let shippingIndexPath = IndexPath(item: inputsRowCount - 1,
+                                      section: PledgeDataSource.Section.inputs.rawValue)
+
+    guard self.indexPathIsShippingLocationCell(shippingIndexPath) else { return nil }
+
+    return shippingIndexPath
+  }
+
+  private func loadProjectSection(delivery: String) {
     self.appendRow(
       value: delivery,
       cellClass: PledgeDescriptionCell.self,
       toSection: Section.project.rawValue
     )
+  }
 
+  private func loadInputsSection(amount: Double, currency: String, currencyCode: String,
+                                 requiresShippingRules: Bool) {
     self.appendRow(
-      value: (amount, currency),
+      value: PledgeInputRow.pledgeAmount(amount: amount, currency: currency),
       cellClass: PledgeAmountCell.self,
       toSection: Section.inputs.rawValue
     )
 
-    self.appendRow(
-      value: (location: "British Indian Ocean Territory", currency: "$", rate: 7.50),
-      cellClass: PledgeShippingLocationCell.self,
-      toSection: Section.inputs.rawValue
-    )
-
-    self.loadSummarySection(isLoggedIn: isLoggedIn)
+    if requiresShippingRules {
+      self.appendRow(
+        value: PledgeInputRow.shippingLocation(location: "", amount: 0.0, currencyCode: currencyCode),
```

Comment:
Why the empty empty values initially?

Reply:
I guess I mean, why not `nil`s?

Reply:
That's a good question, I think visually it's less jarring to see some values that then update than to see nothing and then something 🤷‍♀. In the case of the `location` string, I think you're right that `nil` would probably be better. I'll update that one 👍

Reply:
Going to revisit this one once we do more work on the `PledgeShippingLocationCell`'s view model.